addons/website_sale_comparison/static/src/js/website_sale_comparison.js

```diff
@@ -171,15 +216,17 @@ var ProductComparison = publicWidget.Widget.extend(VariantMixin, {
         $('.o_product_feature_panel').addClass('d-md-block');
         if (!self.comparelist_product_ids.includes(product_id)) {
             self.comparelist_product_ids.push(product_id);
-            if (Object.prototype.hasOwnProperty.call(self.product_data, product_id)) {
-                self._updateContent();
-            } else {
+            self._updateContent();
+            if (!Object.prototype.hasOwnProperty.call(self.product_data, product_id)) {
                 return self._loadProducts([product_id]).then(function () {
                     self._updateContent();
                     self._updateCookie();
                 });
             }
         }
+        else {
+            self._updateContent()
+        }
```

Comment:
@paru-odoo If there is no specific need , we can keep `self._updateContent` out of if...else block.

Reply:
@paru-odoo The way i am looking at it , either way `self._updateContent()` is going to get called,
see according to your code,
 ```js
  if (!self.comparelist_product_ids.includes(product_id)) {
//some code
    self._updateContent()
  }else{
    self._updateContent()
  }
```
so when your if condition fails it will go to else and call it, and vice versa.
Hence keep it out it will work the same.

Reply:
@paru-odoo The way i am looking at it , either way self._updateContent() is going to get called,
see according to your code,

 if (!self.comparelist_product_ids.includes(product_id)) {
//some code
   self._updateContent()
 }else{
   self._updateContent()
 }
so when your if condition fails it will go to else and call it, and vice versa.
Hence keep it out it will work the same.